### DISCUSSION TITLE
garm-cli repo add: tls: failed to verify certificate

### DISCUSSION THREAD

**Discussion by rgl** [cloudbase/garm#290] [Q&A]
while trying to use the ghcr.io/cloudbase/garm:v0.1.5 container image and after running:

```bash
./garm-cli repo add \
  --credentials rgl \
  --owner rgl \
  --name terraform-azure-garm-example-repository \
  --install-webhook \
  --random-webhook-secret
```

the garm logs the following error:

```
time=2024-09-21T19:06:21.151Z level=ERROR msg="installing webhook" error="Get \"https://api.github.com/repos/rgl/terraform-azure-garm-example-repository/hooks?per_page=100\": tls: failed to verify certificate: x509: certificate signed by unknown authority\nfetching hooks\ngithub.com/cloudbase/garm/runner/pool.(*basePoolManager).listHooks
```

inspecting the container image it seems there are no ca-certificates files.

is that expected?

or are we expected to mount the ca-certificates ourselfs?

**Comment by gabriel-samfira** [comment]
*sigh*. We switched to busybox instead of scratch, where we did copy the CA certificates, and I probably forgot about the certificates. I will create a PR in the morning to add those back, but until then, you can definitely mount the host's `/etc/ssl/certs` folder inside the container using `-v /etc/ssl/certs:/etc/ssl/certs:ro`.

**Reply by gabriel-samfira** [reply]
added the certs to the image in this PR:
  * https://github.com/cloudbase/garm/pull/301

The image is building now. It should contain the CA certs once it finishes.
